parquet/src/main/java/org/apache/iceberg/parquet/PruneColumns.java

```diff
@@ -49,7 +49,12 @@ public Type message(MessageType message, List<Type> fields) {
       Type field = fields.get(i);
       Integer fieldId = getId(originalField);
       if (fieldId != null && selectedIds.contains(fieldId)) {
-        builder.addField(originalField);
+        if (field != null) {
+          hasChange = true;
+          builder.addField(field);
+        } else {
+          builder.addField(originalField);
```

Comment:
Should I do the Record Message only thing here as well? where we copy the struct to be empty?

Reply:
What do you mean?